theseus/labs/lie/lie_tensor.py

```diff
@@ -143,26 +337,130 @@ def _no_jop(self, input0: TensorType) -> _JFnReturnType:
     jleft_act = _no_jop
     jleft_project = _no_jop
 
+    def retract(self, delta: TensorType) -> "LieTensor":
+        if not isinstance(delta, torch.Tensor):
+            raise TypeError(
+                "LieTensor.retract() expects a single torch.Tensor argument."
+            )
+        return self.compose(self.exp(delta))
+
+    def local(self, other: "LieTensor") -> torch.Tensor:
+        if not isinstance(other, LieTensor):
+            raise TypeError(
+                f"Incorrect argument for LieTensor.local(). "
+                f"Expected LieTensor, but got {type(other)}."
+            )
+        if not other.ltype == self.ltype:
+            raise ValueError(
+                f"Incorrect ltype for local. Expected {self.ltype}, "
+                f"received {other.ltype}."
+            )
+        return self.inv().compose(other).log()
+
+    # ------------------------------------------------------
+    # Overloaded python and torch operators
+    # ------------------------------------------------------
+    def __add__(self, other: TensorType) -> "LieTensor":
```

Comment:
There was some concern on overloading `+` etc since it could mean different things. Is that resolved? Taosha pointed out that Sophus for instance only overrides `*` and no other operators.

Since we have this stuff added already we can try to get explicit feedback from users on this in the alpha release and then decide if we want to adapt or scale back.

Reply:
Will resolve in example/alpha PR.